app/models/declaration.rb

```diff
@@ -79,4 +81,26 @@ def ineligible_for_funding_reason
       state_reason
     end
   end
+
+  def duplicate_declarations
+    self
+      .class
+      .billable_or_changeable
+      .joins(application: %i[user course])
+      .where(user: { trn: application.user.trn })
+      .where.not(user: { trn: nil })
+      .where.not(user: { id: application.user_id })
+      .where.not(id:)
+      .where(
+        declaration_type:,
+        superseded_by_id: nil,
+        application: { course: application.course.rebranded_alternative_courses },
+      )
+  end
+
+private
+
+  def declaration_date_not_in_the_future
+    errors.add(:declaration_date, :future_declaration_date) if declaration_date&.future?
```

Comment:
in ECF we catch those errors after the validation happens, in NPQ we don't have that so the endpoint will fall over after passing the service validation. Maybe we can move this into the service instead, similar to what has been done here? https://github.com/DFE-Digital/npq-registration/pull/1498/files#diff-f633d59f99e20b9c30469f9dff5e59e7612da3eca47b2055f7b9e1b05004635dR76